src/main/java/gregtech/common/covers/ender/CoverAbstractEnderLink.java

```diff
@@ -0,0 +1,495 @@
+                    .child(new Rectangle()
+                            .setColor(entry.getColor())
+                            .asWidget()
+                            .marginRight(4)
+                            .size(16)
+                            .background(GTGuiTextures.SLOT.asIcon().size(18))
+                            .top(1))
+                    .child(new InteractableText<>(entry, CoverAbstractEnderLink.this::updateColor)
+                            .tooltip(tooltip -> tooltip.setAutoUpdate(true))
+                            .tooltipBuilder(tooltip -> {
+                                String desc = entry.getDescription();
+                                if (!desc.isEmpty())
+                                    tooltip.addLine(desc);
+                            })
+                            .width(64)
+                            .height(16)
+                            .top(1)
+                            .marginRight(4))
+                    .child(new ButtonWidget<>()
+                            .overlay(GuiTextures.GEAR)
+                            .addTooltipLine(IKey.lang("cover.generic.ender.set_description.tooltip"))
+                            .onMousePressed(i -> {
+                                // open entry settings
+                                if (entryDescriptionSH.isPanelOpen()) {
+                                    entryDescriptionSH.closePanel();
+                                } else {
+                                    entryDescriptionSH.openPanel();
+                                }
+                                Interactable.playButtonClickSound();
+                                return true;
+                            }))
+                    .child(createSlotWidget(entry))
+                    .child(new ButtonWidget<>()
+                            .overlay(GTGuiTextures.BUTTON_CROSS)
+                            .setEnabledIf(w -> !Objects.equals(entry.getColor(), activeEntry.getColor()))
+                            .addTooltipLine(IKey.lang("cover.generic.ender.delete_entry"))
+                            .onMousePressed(i -> {
+                                // todo option to force delete, maybe as a popup?
+                                deleteEntry(name);
+                                syncToServer(1, buffer -> NetworkUtils.writeStringSafe(buffer, name));
+                                Interactable.playButtonClickSound();
+                                return true;
+                            }));
+        }
+
+        @Override
+        public void closePanel() {
+            var manager = getSyncManager().getModularSyncManager().getPanelSyncManager(panelName);
+            for (var key : opened) {
+                if (manager.getSyncHandler(key) instanceof PanelSyncHandler psh) {
+                    psh.closePanel();
+                }
+            }
+            super.closePanel();
+        }
+
+        @Override
+        @SuppressWarnings("UnstableApiUsage")
```

Comment:
What's being suppressed here? Do we care about it? May be better to not suppress it and just leave the warning.

Reply:
the supression exists because `closePanelInternal()` is marked as internal.